x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/filters/filters.tsx

```diff
@@ -0,0 +1,348 @@
+                          title={
+                            queryIsValid
+                              ? i18n.translate('xpack.lens.indexPattern.filters.dragToReorder', {
+                                  defaultMessage: 'Drag to reorder',
+                                })
+                              : i18n.translate('xpack.lens.indexPattern.filters.isInvalid', {
+                                  defaultMessage: 'This query is invalid',
+                                })
+                          }
+                        />
+                      </EuiFlexItem>
+                      <EuiFlexItem
+                        grow={true}
+                        data-test-subj="indexPattern-filters-existingFilterContainer"
+                      >
+                        <FilterPopover
+                          isOpenByCreation={idx === localFilters.length - 1 && isOpenByCreation}
+                          setIsOpenByCreation={setIsOpenByCreation}
+                          indexPattern={indexPattern}
+                          filter={filter}
+                          Button={({ onClick }: { onClick: MouseEventHandler }) => (
+                            <EuiLink
+                              className="lnsFiltersOperation__popoverButton"
+                              data-test-subj="indexPattern-filters-existingFilterTrigger"
+                              onClick={onClick}
+                              color={queryIsValid ? 'text' : 'danger'}
+                              title={i18n.translate('xpack.lens.indexPattern.filters.clickToEdit', {
+                                defaultMessage: 'Click to edit',
+                              })}
+                            >
+                              {label || input.query || defaultLabel}
+                            </EuiLink>
+                          )}
+                          setFilter={(f: FilterValue) => {
+                            onChangeValue(f.id, f.input, f.label);
+                          }}
+                        />
+                      </EuiFlexItem>
+                      <EuiFlexItem grow={false}>
+                        <EuiButtonIcon
+                          iconSize="s"
+                          iconType="cross"
+                          color="danger"
+                          data-test-subj="indexPattern-filters-existingFilterDelete"
+                          onClick={() => {
+                            onRemoveFilter(filter.id);
+                          }}
+                          aria-label={i18n.translate(
+                            'xpack.lens.indexPattern.filters.removeSearchQuery',
+                            {
+                              defaultMessage: 'Remove search query',
+                            }
+                          )}
+                          title={i18n.translate('xpack.lens.indexPattern.filters.remove', {
+                            defaultMessage: 'Remove',
+                          })}
```

Comment:
Do you need both aria-label and title?

Reply:
Yes because the `title` helps all users with understand what the `x` will do by showing the browser tooltip, while the `aria-label` is for screen-readers only. But `title` doesn't work the same as `aria-label` for screen-readers so we need both.